### PR TITLE
[Merged by Bors] - feat(Algebra/Homology): the stupid truncation of homological complexes

### PR DESCRIPTION
**Pull request by joelriou**
---
- [x] depends on: #18500

Note about the **large-import** label: I believe we may safely ignore it here. In order to develop the basic theory of truncations, we need to import some bits of homology theory at some point which is what is happening here...

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/Embedding/StupidTrunc.lean [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import Mathlib.Algebra.Homology.Embedding.Extend
+import Mathlib.Algebra.Homology.Embedding.IsSupported
+import Mathlib.Algebra.Homology.Embedding.Restriction
+
+/-!
+# The stupid truncation of homological complexes
+
+Given an embedding `e : c.Embedding c'` of complex shapes, we define
+a functor `stupidTruncFunctor : HomologicalComplex C c' ⥤ HomologicalComplex C c'`
+which sends `K` to `K.stupidTrunc e` which is defined as `(K.restriction e).extend e`.
```
Comment (jcommelin) [inline Mathlib/Algebra/Homology/Embedding/StupidTrunc.lean:16]: I think this means that one of the TODO's in `Mathlib/Algebra/Homology/Embedding/Basic.lean` is now done, on line 33/34.

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Algebra/Homology/Embedding/StupidTrunc.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/Algebra/Homology/Embedding/StupidTrunc.lean:16]: Thanks!

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**